src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class FileLoggerProcessor : IDisposable
+    {
+        private const int _maxQueuedMessages = 1024;
+
+        private readonly string _path;
+        private readonly string _fileName;
+        private readonly int? _maxFileSize;
+        private readonly int? _maxRetainedFiles;
+        private int _fileNumber = 1;
+
+        private readonly BlockingCollection<LogMessage> _messageQueue = new BlockingCollection<LogMessage>(_maxQueuedMessages);
+        private readonly List<LogMessage> _currentBatch = new List<LogMessage>();
+        private Task _outputTask;
+        private CancellationTokenSource _cancellationTokenSource;
+
+        public FileLoggerProcessor(IOptionsMonitor<FileLoggerOptions> options)
```

Comment:
Any reason this is an IOptionsMonitor?

Reply:
Generally you can just use IOptions if you are just using the current value and not holding onto the IOptionsMonitor, since the main benefit from the monitor is the ability updateable values, which you aren't doing if you just look at CurrentValue once.

Reply:
But, if these are going to be created multiple times, where the options might change between requests, IOptionsMonitor is what you want, since IOptions would give you the file logger options that was computed the first time always.

Reply:
Since this is a logging feature it would be desirable for it to respond to config changes. That should at least be filed as a follow-up issue if not implemented in this PR.